pkg/nfd-master/nfd-master.go

```diff
@@ -813,19 +857,120 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.No
 			s.MergeInto(features)
 		}
 
+		// Set the merged features to the NodeFeature object
+		nodeFeatures.Spec = *features
+
 		klog.V(4).InfoS("merged nodeFeatureSpecs", "newNodeFeatureSpec", utils.DelayedDumper(features))
 	}
 
+	return nodeFeatures, nil
+}
+
+func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.Node) error {
+	if m.nfdController == nil || m.nfdController.featureLister == nil {
+		return nil
+	}
+
+	// Merge all NodeFeature objects into a single NodeFeatureSpec
+	nodeFeatures, err := m.mergeNodeFeatures(node.Name)
+	if err != nil {
+		return fmt.Errorf("failed to merge NodeFeature objects for node %q: %w", node.Name, err)
+	}
+
 	// Update node labels et al. This may also mean removing all NFD-owned
 	// labels (et al.), for example  in the case no NodeFeature objects are
 	// present.
-	if err := m.refreshNodeFeatures(cli, node, features.Labels, &features.Features); err != nil {
+	if err := m.refreshNodeFeatures(cli, node, nodeFeatures.Spec.Labels, &nodeFeatures.Spec.Features); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (m *nfdMaster) nfdAPIUpdateAllNodeFeatureGroups() error {
+	klog.V(1).InfoS("updating all NodeFeatureGroups")
+
+	nodeFeatureGroupsList, err := m.nfdClient.NfdV1alpha1().NodeFeatureGroups(m.namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get NodeFeatureGroup objects: %w", err)
+	}
+
+	if len(nodeFeatureGroupsList.Items) > 0 {
+		for _, nodeFeatureGroup := range nodeFeatureGroupsList.Items {
+			m.updaterPool.nfgQueue.Add(nodeFeatureGroup.Name)
+		}
+	} else {
+		klog.V(2).InfoS("no NodeFeatureGroup objects found")
+	}
+
+	return nil
+}
+
+func (m *nfdMaster) nfdAPIUpdateNodeFeatureGroup(nfdClient *nfdclientset.Clientset, nodeFeatureGroup *nfdv1alpha1.NodeFeatureGroup) error {
+	klog.V(1).InfoS("updating NodeFeatureGroup", "nodeFeatureGroup", klog.KObj(nodeFeatureGroup))
```

Comment:
This feels a bit redundant as later we log if the object was updated or (with V(1)) if there were no changes. If we want to keep this, increase the verbosity and change the wording a bit(?)

```suggestion
```

OR

```suggestion
	klog.V(2).InfoS("evaluating NodeFeatureGroup", "nodeFeatureGroup", klog.KObj(nodeFeatureGroup))
```